### PR TITLE
Replace grunt tasks with livestyle/assetgraph based flow

### LINKED ISSUE #453: Atom feed image urls are 404's
Looks like the URL's for images in the Atom feed are not pointing to the revved images:

```tap
not ok 121
  ---
    operator: error
    expected:
      
    actual:
      "Got 404 error retrieving http://yeoman.io/assets/img/blog/tasks.jpg\nIncluding assets:\n    inline Html in http://yeoman.io/blog/atom.xml\n"
    at: http://yeoman.io/assets/img/blog/tasks.jpg
  ...
not ok 122
  ---
    operator: error
    expected:
      
    actual:
      "Got 404 error retrieving http://yeoman.io/assets/img/blog/backbone-app-path.png\nIncluding assets:\n    inline Html in http://yeoman.io/blog/atom.xml\n"
    at: http://yeoman.io/assets/img/blog/backbone-app-path.png
  ...
not ok 123
  ---
    operator: error
    expected:
      
    actual:
      "Got 404 error retrieving http://yeoman.io/assets/img/blog/yo-backbone-mocha.png\nIncluding assets:\n    inline Html in http://yeoman.io/blog/atom.xml\n"
    at: http://yeoman.io/assets/img/blog/yo-backbone-mocha.png
  ...
```

### PR DESCRIPTION
**Pull request by Munter**
Fixes #453 

Added:
- https://github.com/Munter/grunt-livestyle
- https://github.com/Munter/grunt-reduce

Deprecated:
- grunt-autoprefixer
- grunt-contrib-concat
- grunt-contrib-connect
- grunt-contrib-copy
- grunt-contrib-cssmin
- grunt-contrib-htmlmin
- grunt-contrib-imagemin
- grunt-contrib-sass
- grunt-contrib-uglify
- grunt-contrib-watch
- grunt-filerev
- grunt-svgmin
- grunt-usemin

### grunt serve
Jekyll is building in `--watch` mode. All output ends up in `.tmp`. grunt-livestyle runs on port 9000 and serves from `.tmp`. It picks up changes to files and reloads accordingly.

### grunt build
Jekyll does a one-off build into `.tmp`. grunt-reduce pick everything up from here and runs an assetgraph build on it, which ends up in `dist`

This should fix problems with usemin/filerev in general.
I'm sure there will be questions. Ask away!

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -6,14 +6,12 @@
 name: Yeoman.io
 description: Yeoman website
 author:
-  name: Yeoman.io
```
Comment (eddiemonge) [inline _config.yml]: why?

File: app/assets/js/script.js [hunk 2]
```diff
@@ -4,6 +4,11 @@
   var $win = $(window);
   var $doc = $(document);
 
+  // For assetgraph to pick up strings as urls
```
Comment (eddiemonge) [inline app/assets/js/script.js]: I don't really understand what you mean by this. Does AG parse it or something looking for this specific method?

**Comment by SBoudrias** [comment]
This looks pretty good. I'll give it a test over the weekend.

File: app/assets/js/script.js (hunk 2 above)
Comment (Munter) [inline app/assets/js/script.js]: Assetgraph uses static analysis to figure out what files need to go in the build. However, non-module loader file loads, like `$.getJSON('/blacklist.json')` is not something it can recognize as a URL, since it's just a string. So in order for assetgraph to pick it up as a url that should be included in the graph I wrap it in `GETSTATICURL()`.

An alternative way to get assetgraph to pick up `/blacklist.json` in the build would be to name that file explicitly as an initial asset in the `reduce` section of the Gruntfile. This will ensure the file is part of the build, but it won't get revved, since initial assets are treated as files that should be externally linkable.

File: _config.yml (hunk 1 above)
Comment (Munter) [inline _config.yml]: Good question. Must be a mistake. I'll add it back

File: app/assets/js/script.js (hunk 2 above)
Comment (eddiemonge) [inline app/assets/js/script.js]: Could it be done in a comment or something else that gets stripped out in the final build so there isn't JS specifically for the build system in the outputted code?

**Comment by Munter** [comment]
I've uploaded a dist build to http://reduceyeoman.surge.sh/ in case anyone wants to test the output without running the setup themselves

**Comment by sindresorhus** [comment]
This makes me so happy :)

File: app/assets/js/script.js (hunk 2 above)
Comment (Munter) [inline app/assets/js/script.js]: It can be added as a separate script include with `id="bootstrap"`, which the builer will take as an indication to clean it out during build

**Comment by arthurvr** [comment]
:heart_eyes: Lovely work!

File: app/_includes/foot.html [hunk 3]
```diff
@@ -34,17 +34,21 @@
...
 <script src="/assets/js/script.js"></script>
-<!-- endbuild -->
 
 {% if page.title == 'Generators' %}
-  <!-- build:js(app) /assets/js/generators.js -->
+  <script id="bootstrapper">
+      window.GETSTATICURL = function (url) { // , placeHolderValue1, placeHolderValue2, ...
+          var placeHolderValues = Array.prototype.slice.call(arguments, 1);
+          return url.replace(/\*\*?|\{[^\}]*\}/g, function () {
+          return placeHolderValues.shift();
```
Comment (arthurvr) [inline app/_includes/foot.html]: Wrong indentation :)

File: Gruntfile.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ module.exports = function (grunt) {
   require('time-grunt')(grunt);
   // Load all Grunt tasks
   require('load-grunt-tasks')(grunt);
+  grunt.loadNpmTasks('grunt-livestyle');
```
Comment (SBoudrias) [inline Gruntfile.js]: Isn't this loaded by `load-grunt-tasks`?

File: app/_includes/foot.html [hunk 5]
```diff
@@ -34,17 +34,21 @@
...
 
-<!-- build:js(app) /assets/js/scripts.js -->
 <script src="/assets/js/libs/prettify.js"></script>
 <script src="/assets/js/script.js"></script>
-<!-- endbuild -->
 
 {% if page.title == 'Generators' %}
-  <!-- build:js(app) /assets/js/generators.js -->
+  <script id="bootstrapper">
+      window.GETSTATICURL = function (url) { // , placeHolderValue1, placeHolderValue2, ...
```
Comment (SBoudrias) [inline app/_includes/foot.html]: Add a comment explaining why this is required.

File: package.json [hunk 6]
```diff
@@ -2,29 +2,20 @@
...
-    "grunt-contrib-watch": "^0.6.1",
-    "grunt-filerev": "^2.1.2",
     "grunt-jekyll": "^0.4.2",
-    "grunt-svgmin": "^2.0.0",
-    "grunt-usemin": "^3.0.0",
+    "grunt-livestyle": "^1.5.1",
+    "grunt-reduce": "^2.0.0",
     "jshint-stylish": "^1.0.0",
     "load-grunt-tasks": "^3.1.0",
+    "svgo": "^0.4.5",
```
Comment (SBoudrias) [inline package.json]: This dependency is not used anywhere. Why is it required here?

Comment (Munter) [inline package.json]: It's an optional peer dependency of grunt-reduce. If it is available SVG's will be optimized with it.

File: Gruntfile.js (hunk 4 above)
Comment (Munter) [inline Gruntfile.js]: Right. Removing

File: app/_includes/foot.html (hunk 5 above)
Comment (Munter) [inline app/_includes/foot.html]: Done

File: _config.yml [hunk 7]
```diff
@@ -6,14 +6,13 @@
...
+-  name: Yeoman.io
```
Comment (arthurvr) [inline _config.yml]: Why did you add a `-` here? Copy-paste mistake?

Comment (Munter) [inline _config.yml]: Most likely. Fixing :)

**Comment by SBoudrias** [comment]
LGTM, let's squash - fix merge conflict and release :)

**Comment by eddiemonge** [comment]
merging, rebasing, final testing and then pushing
